apps/web/app/(app)/environments/[environmentId]/settings/lookandfeel/actions.ts

```diff
@@ -5,14 +5,21 @@ import { updateProduct } from "@formbricks/lib/product/service";
 import { TProductUpdateInput } from "@formbricks/types/product";
 import { getServerSession } from "next-auth";
 import { AuthorizationError } from "@formbricks/types/errors";
-import { canUserAccessProduct } from "@formbricks/lib/product/auth";
+import { canUserAccessProduct, verifyUserRoleAccess } from "@formbricks/lib/product/auth";
 
-export async function updateProductAction(productId: string, inputProduct: Partial<TProductUpdateInput>) {
+export async function updateProductAction(
+  productId: string,
+  inputProduct: Partial<TProductUpdateInput>,
+  environmentId: string
+) {
   const session = await getServerSession(authOptions);
   if (!session) throw new AuthorizationError("Not authorized");
 
   const isAuthorized = await canUserAccessProduct(session.user.id, productId);
   if (!isAuthorized) throw new AuthorizationError("Not authorized");
 
+  const { hasCreateOrUpdateAccess } = await verifyUserRoleAccess(environmentId, session.user.id);
```

Comment:
we have the same action problem as mentioned below. Please derive environmentId instead of param.